src/app/file-manager/file-modal/file-modal.component.ts

```diff
@@ -1,43 +1,104 @@
-import {Component, Inject} from '@angular/core';
+import {Component, Inject, Output, EventEmitter} from '@angular/core';
 import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from '@angular/material';
+import {TreeNode} from 'primeng/primeng';
+import { Item } from '@app/file-manager/models/item';
+import * as Downloadables from '../actions/downloadables.actions';
+import { Store } from '@ngrx/store';
+import { AppState } from '@app/file-manager/transferables-grid/transferables-grid.component';
+import { TransferablesGridComponent } from '../transferables-grid/transferables-grid.component';
+import { Message } from 'primeng/components/common/api';
+import { Status } from '@app/file-manager/models/status';
 
 @Component({
   selector: 'app-file-modal',
   templateUrl: './file-modal.component.html',
   styleUrls: ['./file-modal.component.scss']
 })
 export class FileModalComponent  {
+  @Output('done') done: EventEmitter<any> = new EventEmitter();
 
   preserveStructure = true;
   directory = 'Choose Directory...';
   isValid = false;
+  dataFile: Item;
```

Comment:
It looks like `dataFile` is only used in `setItems()`, if so it should be changed to a local variable.

Reply:
`dataFile` is now a local variable.